src/main/scala/com/target/data_validator/ValidatorTable.scala

```diff
@@ -210,7 +211,18 @@ case class ValidatorHiveTable(
 
   override def getDF(session: SparkSession): Try[DataFrame] = {
     logger.info(s"Opening table: $db.$table")
-    Try(session.table(s"$db.$table"))
+    useHWC match {
+      case Some(x) if x =>
+        val hive = com.hortonworks.hwc.HiveWarehouseSession.session(session).build()
+        val query = condition match {
```

Comment:
Yes, I should have commented.  I intentionally did it this way.  

[Hortonworks recommends disabling the pushdown functionality](https://docs.cloudera.com/HDPDocuments/HDP3/HDP-3.1.4/integrating-hive/content/hive-read-write-operations.html) because it is broken with HWC.  So there is no query plan optimization being done.  So if we don't filter here, as best I can tell in my experiments, the DataSourceV2 reader attempts to read the entire table.  [This SO](https://stackoverflow.com/questions/59758860/how-can-we-safely-reuse-a-dataframe-reading-a-hive-table-using-hivewarehouseconn) kinda summarizes things as well.

So if you accept the above, then the question becomes should we make an effort to remove the redundant filter operation in the open method for the `useHWC` case?

Reply:
In that case, as you mentioned, we should get rid of the redundant filtering when `useHWC` is `true`.

And we need to take care of [this](https://github.com/phpisciuneri/data-validator/blob/d0608cdd4be88ed054c713a585c6d141f6aa7aa7/src/main/scala/com/target/data_validator/ConfigVar.scala#L78) as well.

Reply:
@samratmitra-0812 see 3eaaf18 for the fix to the redundant filtering in the `open` method.

Reply:
@phpisciuneri This commit looks good to me.

And as I mentioned earlier, we need to take care of [this](https://github.com/phpisciuneri/data-validator/blob/d0608cdd4be88ed054c713a585c6d141f6aa7aa7/src/main/scala/com/target/data_validator/ConfigVar.scala#L78) as well.

Reply:
@samratmitra-0812 see 457fd2c.  

Haven't started working on documentation yet because I wanted to wait until we are set on the details of this PR, but again config would support an optional `useHWC` parameter:

```yaml
vars:
  - name: MAX_AGE
    useHWC: true
    sql: select max(age) from census_income.adult
```

Reply:
@phpisciuneri LGTM.

I was actually thinking of defining the `usHWC` flag once at `ValidatorConfig` level, instead of having to define it multiple times. But that would require lot of modifications to the existing code. So I am OK with the current implementation. May be we should think about refactoring this later.